### PR TITLE
Format Writing InSpec profiles (toRuby function)

### PR DESCRIPTION
**Pull request by em-c-rod**
The main goal of this PR is to make the writing of InSpec profiles (for use in things like stub generation and delta) more consistent with cookstyle formatting for ease of use when comparing with new changes from delta and when generating InSpec stubs that match a standard format.
When using this library to parse InSpec profiles or xccdf files and generate InSpec profiles, the general workflow is as follows: 
The input is processed, read into a typescript object, operated on with any required action / logic, then written into an InSpec profile as output. This means that we can not simply write out in the same format we got in. Instead, we have to make choices about formatting for how to write out content. 
Some of the changes below do make the formatting more consistent with cookstyle, but it is not 100% the same, nor does cookstyle cover every choice we have to make (ex: `tag key: 'value'` vs. `tag 'key': 'value'` is not covered in cookstyle)
- [x] Eliminate redundant "default" describe value when writing the profile. There was an issue when running the toRuby function where it would write `desc: blah` and `desc: 'default': blah` which was redundant and not needed.
- [x] Change several things to have single quotes instead of double quotes to match cookstyle preferences
- [x] Change the functionality so that the "# encoding: UTF-8\n\n" string is not put on each control.
- [x] Add the describe block to the json parser. Before, it was only populating the metadata.
- [x] Add test for control `toRuby` function
- [x] change complicated string outputs to use %q() format instead of ""

### CODE REVIEW

**Comment by ejaronne** [comment]
Questions: 
Does this preserve the exact original content from the guidance document (STIG, CIS, Vendor)? For example, command-line instructions from the author may have very specific punctuation that needs to be preserved exactly as intended.

Does it also not artificially wrap text (especially check and fix text), which may already have carriage returns and/or linefeeds in the original content?  I know we've struggled over the years to ensure stuff looks decent in Heimdall, csv, cli, CAAT, ckl.

File: src/objects/control.ts [hunk 1]
```diff
@@ -103,9 +103,14 @@ export default class Control {
     if (this.descs) {
       Object.entries(this.descs).forEach(([key, desc]) => {
         if (desc) {
-          result += `  desc "${key}", "${escapeDoubleQuotes(
-            removeNewlinePlaceholders(desc)
-          )}"\n`;
+          if(key.match("default") && this.desc) {
+            console.error(`${this.id} has a redundant default value. This command will write the "desc" value to the control and will not write the "default" value.`);
```
Comment (aaronlippold) [inline src/objects/control.ts:107]: This is what is expected. The new standard is to use the .descriptions.... object , the main desc object is there for backwards compatibility.

Comment (aaronlippold) [reply src/objects/control.ts:107]: you can review the profile schema or the ruby inspec-objects as a reference if needed in the future.

File: test/tests/control.spec.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+describe('The control functionality', () => {
+    const cookstyle_profile = processInSpecProfile(fs.readFileSync('test/sample_data/inspec/json/cookstyle-controls-profile.json', 'utf-8'))
+
+    cookstyle_profile.controls.forEach(control => {
+        // Write the new control to the controls folder
+        fs.writeFileSync(path.join('test/sample_data/', 'controls-test-results', `${control.id}.rb`), control.toRuby())
+    })
+    
+    // This check is comparing what the function "toRuby" is outputting with a small sample profile that has been formatted
```
Comment (aaronlippold) [inline test/tests/control.spec.ts:13]: Please start the TSDoc process here and expand it out at least as a simple lift and shift and start the minimal GitHub Action to generate the TSDocs - <http://tsdocs.org>

**Review by aaronlippold — CHANGES_REQUESTED** [review]

**Review by aaronlippold — APPROVED** [review]

**Pull request merged**